mas-foundation/src/main/java/com/ca/mas/core/storage/sharedstorage/AccountManagerUtil.java

```diff
@@ -51,27 +58,42 @@ public AccountManagerUtil(Context context, String accountName, boolean sharedSto
             mAccountManager = AccountManager.get(MAS.getContext());
             //Attempt to retrieve the account
             Account[] accounts = mAccountManager.getAccountsByType(accountType);
+            messageBuilder.append(" existing accounts (" + accountType + ")=" + accounts.length);
             for (Account account : accounts) {
-                if (accountName.equals(account.name)) {
-                    String password = mAccountManager.getPassword(account);
-                    String savedPassword = identifier.toString();
-                    if (password.equals(savedPassword)) {
-                        mAccount = account;
-                    }else {
-                        // - case migration from old AccountManagerStoreDataSource
-                        mAccount = null;
-                        identifier = new SharedStorageIdentifier();
+                messageBuilder.append(" trying account:" + account.name);
+                synchronized (this) {
```

Comment:
Use class-object for synchronization.
synchronization (AccountManagerUtil.class) {
}

Reply:
Can I use the object already created (mutex)?